src/table_builder.h

```diff
@@ -49,6 +52,7 @@ class TitanTableBuilder : public TableBuilder {
   std::unique_ptr<BlobFileHandle> blob_handle_;
   std::shared_ptr<BlobFileManager> blob_manager_;
   std::unique_ptr<BlobFileBuilder> blob_builder_;
+  std::weak_ptr<VersionSet> vset_ref_;
```

Comment:
what's the rationale to use weak_ptr?

Reply:
Originally I was trying to make it comfortable to pass a null pointer as vset in table-builder test. And also try not to make temporary table-builder instance hold strong reference.